### ISSUE TITLE
(macOS M1) Bootstrapping gnupg fails

### ISSUE THREAD

**Issue by brechtm** [LLNL/spack#35339]
### Steps to reproduce

Since I'm happily using spack on a set of Linux servers, I'm evaluating it for use on my Mac to replace Homebrew (I'd like to maintain installations of specific package versions).

Unfortunately, bootstrapping gnupg fails:

```console
% spack mirror add binary_mirror  https://binaries.spack.io/develop
% spack buildcache keys --install --trust
```

P.S. Is spack on M1/M2 Macs considered stable enough to depend on for daily use?

### Error message

```console
==> Fetching https://binaries.spack.io/develop/build_cache/_pgp/2C8DD3224EF3573A42BD221FA8E0CA3C1C2ADA2F.pub
==> Bootstrapping gnupg from pre-built binaries
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-m1-apple-clang-13.1.6-zlib-1.2.13-azrxnl6yp7xeapfy7nljiopucaelofuh.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-m1/apple-clang-13.1.6/zlib-1.2.13/darwin-monterey-m1-apple-clang-13.1.6-zlib-1.2.13-azrxnl6yp7xeapfy7nljiopucaelofuh.spack
==> Installing "zlib@1.2.13%apple-clang@13.1.6+optimize+pic+shared build_system=makefile arch=darwin-monterey-m1" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-libiconv-1.16-id44zneq3nh4grvtekqoefl24okct4ak.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/libiconv-1.16/darwin-monterey-aarch64-apple-clang-13.1.6-libiconv-1.16-id44zneq3nh4grvtekqoefl24okct4ak.spack
==> Installing "libiconv@1.16%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools libs=shared,static arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-npth-1.6-lp7fobvpwlk3xugo7th2kmcnrvqqxb3b.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/npth-1.6/darwin-monterey-aarch64-apple-clang-13.1.6-npth-1.6-lp7fobvpwlk3xugo7th2kmcnrvqqxb3b.spack
==> Installing "npth@1.6%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-libassuan-2.5.5-trhqsquxpocecfgkeif5bh2dwgu4njbp.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/libassuan-2.5.5/darwin-monterey-aarch64-apple-clang-13.1.6-libassuan-2.5.5-trhqsquxpocecfgkeif5bh2dwgu4njbp.spack
==> Installing "libassuan@2.5.5%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Warning: libgpg-error@1.46%apple-clang@13.1.6 build_system=autotools arch=darwin-monterey-aarch64/yg67voz is being registered in the database with prefix /Users/brechtm/.spack/bootstrap/store/darwin-monterey-aarch64/apple-clang-13.1.6/libgpg-error-1.46-yg67vozcaac75p3dnhd6c3cjpa5nsfjo, but this directory does not contain an installation of the spec, due to: Install prefix /Users/brechtm/.spack/bootstrap/store/darwin-monterey-aarch64/apple-clang-13.1.6/libgpg-error-1.46-yg67vozcaac75p3dnhd6c3cjpa5nsfjo does not exist.
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-libgcrypt-1.10.1-eadvdhou2xjdhf47x3q5x2ypa4qhfqjy.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/libgcrypt-1.10.1/darwin-monterey-aarch64-apple-clang-13.1.6-libgcrypt-1.10.1-eadvdhou2xjdhf47x3q5x2ypa4qhfqjy.spack
==> Installing "libgcrypt@1.10.1%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-libgpg-error-1.46-yg67vozcaac75p3dnhd6c3cjpa5nsfjo.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/libgpg-error-1.46/darwin-monterey-aarch64-apple-clang-13.1.6-libgpg-error-1.46-yg67vozcaac75p3dnhd6c3cjpa5nsfjo.spack
==> Installing "libgpg-error@1.46%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-libksba-1.6.2-m7o6qwsu2gxvlka2jbd5puzwj3z553ob.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/libksba-1.6.2/darwin-monterey-aarch64-apple-clang-13.1.6-libksba-1.6.2-m7o6qwsu2gxvlka2jbd5puzwj3z553ob.spack
==> Installing "libksba@1.6.2%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-pinentry-1.2.1-6m36xv6ft3yterimp6xoozz66ych5eew.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/pinentry-1.2.1/darwin-monterey-aarch64-apple-clang-13.1.6-pinentry-1.2.1-6m36xv6ft3yterimp6xoozz66ych5eew.spack
==> Installing "pinentry@1.2.1%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools gui=tty arch=darwin-monterey-aarch64" from a buildcache
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64-apple-clang-13.1.6-gnupg-2.3.8-pyrfgqkgltgfk4yljfw2myxn6vqen2j6.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.4/build_cache/darwin-monterey-aarch64/apple-clang-13.1.6/gnupg-2.3.8/darwin-monterey-aarch64-apple-clang-13.1.6-gnupg-2.3.8-pyrfgqkgltgfk4yljfw2myxn6vqen2j6.spack
==> Installing "gnupg@2.3.8%apple-clang@13.1.6 cflags="-mmacosx-version-min=10.13"  build_system=autotools arch=darwin-monterey-aarch64" from a buildcache
==> Error: Command exited with status -9:
'/Users/brechtm/.spack/bootstrap/store/darwin-monterey-aarch64/apple-clang-13.1.6/gnupg-2.3.8-pyrfgqkgltgfk4yljfw2myxn6vqen2j6/bin/gpg2' '--version'
```

### Information on your system

* **Spack:** 0.19.0 (a86911246a8c5571ca210ff25fc990b8eb297a5c)
* **Python:** 3.9.6
* **Platform:** darwin-monterey-m1
* **Concretizer:** clingo

I don't know what it means to *run the failing commands in debug mode* as suggested in the issue template. Please specify this in the template. I have not found a verbose switch to pass to spack.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
I suspect it's the same issue as #32571

**Comment by alalazo** [comment]
This should be fixed by #35585
